src/main/java/com/health/threadpool/TaskProcessingService.java

```diff
@@ -53,9 +65,319 @@ public class TaskProcessingService {
+                requestType, language, view_url, documentId, documentPath, documentType, outlinePath);
+
+        if (documentPath == null) {
+
+            resultMap.put(CommonData.STATUS, CommonData.STATUS_FAILED);
+            resultMap.put(CommonData.REASON, "documentPath file does not exist");
+            return resultMap;
+
+        }
+
+        if (outlinePath != null && !outlinePath.isPresent()) {
+
+            resultMap.put(CommonData.STATUS, CommonData.STATUS_FAILED);
+            resultMap.put(CommonData.REASON, "outline file does not exist");
+            return resultMap;
+
+        }
+
+        QueueManagement queuemnt = new QueueManagement();
+
+        if (outlinePath != null && outlinePath.isPresent())
+            queuemnt.setOutlinePath(outlinePath.get());
+        queuemnt.setRequestTime(getCurrentTime());
+        queuemnt.setRequestType(requestType);
+        queuemnt.setDocumentId(documentId);
+        queuemnt.setDocumentType(documentType);
+        queuemnt.setDocumentPath(documentPath);
+        queuemnt.setDocumentUrl(documentUrl);
+        queuemnt.setRank(rank);
+        queuemnt.setViewUrl(view_url);
+        queuemnt.setLanguageId(languageId);
+        queuemnt.setStatus("pending");
+        if (language != null)
+            queuemnt.setLanguage(language);
+        if (category != null && category.isPresent())
+            queuemnt.setCategory(category.get());
+        if (categoryId != null && categoryId.isPresent())
+            queuemnt.setCategoryId(categoryId.get());
+        if (topic != null && topic.isPresent())
+            queuemnt.setTopic(topic.get());
+        if (topicId != null && topicId.isPresent())
+            queuemnt.setTopicId(topicId.get());
+
+        queueRepo.save(queuemnt);
+
+        resultMap.put(CommonData.QUEUE_ID, Long.toString(queuemnt.getQueueId()));
+        resultMap.put(CommonData.STATUS, CommonData.SUCCESS);
+
+        return resultMap;
+
+    }
+
+    public void addUpdateDeleteTutorial(Tutorial tutorial, String requestType) {
+
+        String documentId = CommonData.DOCUMENT_TYPE_TUTORIAL + tutorial.getTutorialId();
+        String documentType = CommonData.DOCUMENT_TYPE_TUTORIAL;
+        String documentPath = tutorial.getTimeScript();
+        String documentUrl = "";
+        if (tutorial.getConAssignedTutorial().getLan().getLanId() == 22)
+            documentUrl = "/TimeScript/" + tutorial.getTutorialId();
+        else
+            documentUrl = "/OriginalScript/" + tutorial.getTutorialId();
+
+        int rank = tutorial.getUserVisit() + 3 * tutorial.getResourceVisit();
+        String view_url = null;
+        int languageId = tutorial.getConAssignedTutorial().getLan().getLanId();
+        String languag = tutorial.getConAssignedTutorial().getLan().getLangName();
+        Optional<Integer> categoryId = Optional
+                .of(tutorial.getConAssignedTutorial().getTopicCatId().getCat().getCategoryId());
```

Comment:
```suggestion
        TopicCategoryMapping topicCat = conAssignedTutorial.getTopicCatId();
        Optional<Integer> categoryId = Optional.of(topicCat.getCat().getCategoryId());
```
Assign to a variable.

Also, use above variable.